From 743ef168b8dbf3c860bc58192fd0fc51afe7bfbe Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Tue, 6 Aug 2024 15:34:06 -0700
Subject: [PATCH 4/4] giflib files does not build with c-extension

pytorch_vision registers them as a CppExtension files
in setup.py and this causes the pytorch builder
to add the -std=c++17 build flag. That flag
will then cause build break at least on
mac, Fedora 40 and Mageia 9.

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 setup.py                                                        | 2 +-
 .../csrc/io/image/cpu/giflib/{dgif_lib.c => dgif_lib.cpp}       | 0
 .../csrc/io/image/cpu/giflib/{gif_hash.c => gif_hash.cpp}       | 0
 .../csrc/io/image/cpu/giflib/{gifalloc.c => gifalloc.cpp}       | 0
 .../giflib/{openbsd-reallocarray.c => openbsd-reallocarray.cpp} | 0
 5 files changed, 1 insertion(+), 1 deletion(-)
 rename torchvision/csrc/io/image/cpu/giflib/{dgif_lib.c => dgif_lib.cpp} (100%)
 rename torchvision/csrc/io/image/cpu/giflib/{gif_hash.c => gif_hash.cpp} (100%)
 rename torchvision/csrc/io/image/cpu/giflib/{gifalloc.c => gifalloc.cpp} (100%)
 rename torchvision/csrc/io/image/cpu/giflib/{openbsd-reallocarray.c => openbsd-reallocarray.cpp} (100%)

diff --git a/setup.py b/setup.py
index dd032c942f..3be4b47cad 100644
--- a/setup.py
+++ b/setup.py
@@ -270,7 +270,7 @@ def make_image_extension():
     define_macros, extra_compile_args = get_macros_and_flags()
 
     image_dir = CSRS_DIR / "io/image"
-    sources = list(image_dir.glob("*.cpp")) + list(image_dir.glob("cpu/*.cpp")) + list(image_dir.glob("cpu/giflib/*.c"))
+    sources = list(image_dir.glob("*.cpp")) + list(image_dir.glob("cpu/*.cpp")) + list(image_dir.glob("cpu/giflib/*.cpp"))
 
     if IS_ROCM:
         sources += list(image_dir.glob("hip/*.cpp"))
diff --git a/torchvision/csrc/io/image/cpu/giflib/dgif_lib.c b/torchvision/csrc/io/image/cpu/giflib/dgif_lib.cpp
similarity index 100%
rename from torchvision/csrc/io/image/cpu/giflib/dgif_lib.c
rename to torchvision/csrc/io/image/cpu/giflib/dgif_lib.cpp
diff --git a/torchvision/csrc/io/image/cpu/giflib/gif_hash.c b/torchvision/csrc/io/image/cpu/giflib/gif_hash.cpp
similarity index 100%
rename from torchvision/csrc/io/image/cpu/giflib/gif_hash.c
rename to torchvision/csrc/io/image/cpu/giflib/gif_hash.cpp
diff --git a/torchvision/csrc/io/image/cpu/giflib/gifalloc.c b/torchvision/csrc/io/image/cpu/giflib/gifalloc.cpp
similarity index 100%
rename from torchvision/csrc/io/image/cpu/giflib/gifalloc.c
rename to torchvision/csrc/io/image/cpu/giflib/gifalloc.cpp
diff --git a/torchvision/csrc/io/image/cpu/giflib/openbsd-reallocarray.c b/torchvision/csrc/io/image/cpu/giflib/openbsd-reallocarray.cpp
similarity index 100%
rename from torchvision/csrc/io/image/cpu/giflib/openbsd-reallocarray.c
rename to torchvision/csrc/io/image/cpu/giflib/openbsd-reallocarray.cpp
-- 
2.45.2

